webpack.config.js

```diff
@@ -1,60 +1,71 @@
 const env = process.env.NODE_ENV || 'development';
-
-const webpack = require('webpack');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
 const path = require('path');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin')
-const webpackUMDExternal = require('webpack-umd-external');
+
+const isProduction = env === 'production';
+
+const PATHS = {
+  root: path.join(__dirname, ''),
+  example: path.join(__dirname, 'example')
+};
 
 const pluginsList = [];
-const outputFileName = env === 'production' ?
-  'react-star-rating-component.min.js' :
-  'react-star-rating-component.js';
-
-if (env === 'production') {
-  pluginsList.push(
-    new UglifyJsPlugin({
-      uglifyOptions: {
-        compress: { warnings: false },
-        output: { comments: false }
-      }
-    })
-  );
-}
 
 const config = {
   mode: env,
-
-  entry: path.join(__dirname, 'src/StarRatingComponent.jsx'),
+  entry: {
+    'dist/react-star-rating-component': './index.jsx',
+    'example/bundle': `${PATHS.example}/index.jsx`
+  },
 
   output: {
-    path: path.join(__dirname, 'dist'),
-    filename: outputFileName,
-    library: 'ReactStarRatingComponent',
+    path: PATHS.root,
+    filename: `[name]${isProduction ? '.min' : ''}.js`,
     libraryTarget: 'umd',
     umdNamedDefine: true
   },
 
-  externals: webpackUMDExternal({
-    'react': 'React'
-  }),
+  externals: {
+    'react': 'React',
+    'react-dom': 'ReactDOM'
+  },
 
   resolve: {
     extensions: ['.js', '.jsx']
   },
 
+  optimization: {
+    minimizer: [
+      new UglifyJsPlugin({
```

Comment:
will it minify code for non-`.min` version too?

Reply:
Yes, using Webpack's new "mode" functionality, the minification will only happen if NODE_ENV === 'production', as well as other optimizations specific to development/production environments. See this Medium article for more info: https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a